dialogue-core/src/main/java/com/palantir/dialogue/core/TracedChannel.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.dialogue.Channel;
+import com.palantir.dialogue.Endpoint;
+import com.palantir.dialogue.Request;
+import com.palantir.dialogue.Response;
+import com.palantir.tracing.Tracers;
+
+public final class TracedChannel implements Channel {
+    private final Channel delegate;
+    private final String operationName;
+
+    TracedChannel(Channel delegate, String operationName) {
```

Comment:
I think this either needs a public factory, or the class should be package private

Reply:
made package private to match other *Channel classes.